azurerm/resource_arm_netapp_volume_test.go

```diff
@@ -0,0 +1,465 @@
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	resourceName := "azurerm_netapp_volume.test"
+	ri := tf.AccRandTimeInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMNetAppVolumeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMNetAppVolume_basic(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNetAppVolumeExists(resourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMNetAppVolume_requiresImport(ri, testLocation()),
+				ExpectError: testRequiresImportError("azurerm_netapp_volume"),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMNetAppVolume_complete(t *testing.T) {
+	resourceName := "azurerm_netapp_volume.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMNetAppVolumeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMNetAppVolume_complete(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNetAppVolumeExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "service_level", "Premium"),
+					resource.TestCheckResourceAttr(resourceName, "usage_threshold", "101"),
+					resource.TestCheckResourceAttr(resourceName, "export_policy_rule.#", "1"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMNetAppVolume_update(t *testing.T) {
+	resourceName := "azurerm_netapp_volume.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMNetAppVolumeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMNetAppVolume_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNetAppVolumeExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "usage_threshold", "100"),
+					resource.TestCheckResourceAttr(resourceName, "export_policy_rule.#", "0"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAzureRMNetAppVolume_complete(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNetAppVolumeExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "usage_threshold", "101"),
+					resource.TestCheckResourceAttr(resourceName, "export_policy_rule.#", "1"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
```

Comment:
Could we have another step going back to basic?

Reply:
fixed